### PR TITLE
Python 3 compatibility

### PR DESCRIPTION
**Pull request by frenzymadness**
Hello.

This PR is here for you so you can follow my work on Python 3 compatibility for Bodhi.

I've made a lot of changes to bring some Python 3 compatibility to the codebase and keep passing all tests with Python 2.

I need to solve some dependencies issues before I can run tests with Python 3 so expect some more patches coming soon when dependencies will be ready.

In the meantime, feel free to do a review of existing changes and suggest anything that can be done better.

Have a nice day.
Lumír

### CODE REVIEW

**Comment by frenzymadness** [comment]
CI failure is related to insufficient test coverage of new code but actually, there is no new code just rewritten code to compatible form. Let's fix this later.

File: bodhi/server/captcha.py
Comment (bowlofeggs) [inline bodhi/server/captcha.py:18]: division… so futuristic!

File: bodhi/tests/client/test___init__.py
Comment (bowlofeggs) [inline bodhi/tests/client/test___init__.py]: This comment might be the result of a merge conflict - it should still say the former string:

```
"""Assert that we use init_username if --user is not given."""
```

Bodhi no longer uses the ```USERNAME``` env var.

File: bodhi/tests/server/consumers/test_masher.py
Comment (bowlofeggs) [inline bodhi/tests/server/consumers/test_masher.py]: PEP-8 import style would group ```six``` with the other third party imports, like ```mock``` above.

File: bodhi/tests/server/services/test_updates.py
Comment (bowlofeggs) [inline bodhi/tests/server/services/test_updates.py]: Same here and elsewhere.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]
Thanks so much, this all looks great! I only have a couple minor comments.

As for the missing test coverage, I agree with you but I'm gonna take a look at how much coverage we're talking about. If it's not too much, I might just write a few PRs to cover the missing lines. I'll report back about that shortly!

**Comment by bowlofeggs** [comment]
It looks like we only need coverage on three lines, so I'm just gonna go and write some tests. Once those are merged, we can rebase this and we should get the tests to fully pass. Thanks again for this patch!

**Comment by bowlofeggs** [comment]
#2011 Adds tests to hit those missing lines in ```bodhi.server.bugs```. I realized when writing it that you will need to refactor one of the tests after rebasing since it raises the xmlrpclib.Fault, but that should be pretty easy.

**Comment by bowlofeggs** [comment]
It turns out that the third missing line was in a function that was unused, so I just decided to remove that function rather than test it, in https://github.com/fedora-infra/bodhi/pull/2012

**Comment by bowlofeggs** [comment]
I've confirmed that my two patches merged with yours does get that coverage issue to go away. I'll ask for quick review on those so we can proceed with this PR :)

**Comment by bowlofeggs** [comment]
Alright, if you rebase this patch on top of ```develop```, the coverage should pass.

File: bodhi/tests/client/test___init__.py
Comment (frenzymadness) [reply bodhi/tests/client/test___init__.py]: Yes, damn merge conflict. Fixed.

File: bodhi/tests/server/consumers/test_masher.py
Comment (frenzymadness) [reply bodhi/tests/server/consumers/test_masher.py]: Didn't know that, thanks. Fixed in the latest commit.

**Comment by frenzymadness** [comment]
I've addressed all comments, rebased branch and refactored mentioned test (with Fault exception from xmlrpc_client).

**Comment by frenzymadness** [comment]
I've rebased this PR to the top of current develop branch.

Next steps (just for the record):
* New release of pyramid-fas-openid (with Python 3)
* Conditions of PackageDB
* Bodhi tests with Python 3
* Problem solving

**Comment by bowlofeggs** [comment]
I just rebased this PR to resolve a few conflicts I introduced in another PR. Unfortunately, our CI will fail today because Fedora's container registry is not available due to a scheduled maintenance outage. We should be able to merge this in a few days once it is back online.

**Review by bowlofeggs — APPROVED** [review]
Thanks!

**Comment by bowlofeggs** [comment]
jenkies test

File: bodhi/server/models.py
Comment (bowlofeggs) [inline bodhi/server/models.py]: The test failure is due to removing this blank line:

```
f27	bodhi/server/models.py:155 in public class `DeclEnum`:
f27	        D204: 1 blank line required after class docstring (found 0)
```

It's an easy fix, so I'll take care of it for you.

**Review by bowlofeggs — CHANGES_REQUESTED** [review]

**Review by bowlofeggs — APPROVED** [review]
Thanks!

**Comment by frenzymadness** [comment]
Thanks a lot for all your work and help! I'll create an issue for tracking the final phase of Python 3 compatibility.

**Pull request merged**
